### PR TITLE
Remove NO_PYTHON macros from Exceptions.h/cpp

### PR DESCRIPTION
**Pull request by zrphercule**
This is the live PR of removing NO_PYTHON macros from Exceptions.h/cpp.

### CODE REVIEW

File: torch/csrc/autograd/engine.cpp [hunk 1]
```diff
@@ -478,9 +478,9 @@ auto Engine::execute(const edge_list& input_roots,
 }
 
 #ifdef NO_PYTHON
-Engine& Engine::get_default_engine() {
-  static Engine engine;
-  return engine;
+Engine& Engine::get_default_engine(){
```
Comment (zdevito) [inline torch/csrc/autograd/engine.cpp]: Can you revert this formatting change?

File: torch/csrc/autograd/python_engine.cpp [hunk 2]
```diff
@@ -27,7 +27,6 @@ static torch::autograd::python::PythonEngine engine;
 // throughout the code in both NO_PYTHON builds and regular builds
 Engine& torch::autograd::Engine::get_default_engine() {
   return engine;
-}
```
Comment (zdevito) [inline torch/csrc/autograd/python_engine.cpp]: Missing bracket is causing all the tests to fail.

File: torch/csrc/autograd/engine.cpp (hunk 1 above)
Comment (zrphercule) [reply torch/csrc/autograd/engine.cpp]: sure

File: torch/csrc/autograd/python_engine.cpp (hunk 2 above)
Comment (zrphercule) [reply torch/csrc/autograd/python_engine.cpp]: yes, I mistakely deleted it when resolving conflict

**Review by ezyang — APPROVED** [review]
Whoo deleting code

**Comment by ezyang** [comment]
@zrphercule does Live PR mean you want this merged or not?

**Pull request merged**